core/src/main/java/org/elasticsearch/rest/action/support/RestActions.java

```diff
@@ -85,7 +89,7 @@ public static void buildBroadcastShardsHeader(XContentBuilder builder, ToXConten
         builder.endObject();
     }
 
-    public static QuerySourceBuilder parseQuerySource(RestRequest request) {
+    public static QueryBuilder<?> parseQuerySource(RestRequest request) {
```

Comment:
Does it make sense to remove this method too while we are touching it? I don't think it parses anything, its name should reflect what it does.